tensorflow/python/data/util/random_seed_test.py

```diff
@@ -18,64 +18,120 @@
+      )
+  ]
+  def reduce_fn(x, y):
+    name, input_fn, output_fn = y
+    return x + combinations.combine(
+        input_fn=combinations.NamedObject(
+            "input_fn.{}".format(name), input_fn),
+        output_fn=combinations.NamedObject(
+            "output_fn.{}".format(name), output_fn)
+    )
+
+  return functools.reduce(reduce_fn, cases, [])
+
+class RandomSeedTest(test_base.DatasetTestBase, parameterized.TestCase):
 
-  @test_util.run_in_graph_and_eager_modes
-  def testRandomSeed(self):
-    zero_t = constant_op.constant(0, dtype=dtypes.int64, name='zero')
-    one_t = constant_op.constant(1, dtype=dtypes.int64, name='one')
-    intmax_t = constant_op.constant(
-        2**31 - 1, dtype=dtypes.int64, name='intmax')
-    test_cases = [
-        # Each test case is a tuple with input to get_seed:
-        # (input_graph_seed, input_op_seed)
-        # and output from get_seed:
-        # (output_graph_seed, output_op_seed)
-        ((None, None), (0, 0)),
-        ((None, 1), (random_seed.DEFAULT_GRAPH_SEED, 1)),
-        ((1, 1), (1, 1)),
-        ((0, 0), (0, 2**31 - 1)),  # Avoid nondeterministic (0, 0) output
-        ((2**31 - 1, 0), (0, 2**31 - 1)),  # Don't wrap to (0, 0) either
-        ((0, 2**31 - 1), (0, 2**31 - 1)),  # Wrapping for the other argument
-        # Once more, with tensor-valued arguments
-        ((None, one_t), (random_seed.DEFAULT_GRAPH_SEED, 1)),
-        ((1, one_t), (1, 1)),
-        ((0, zero_t), (0, 2**31 - 1)),  # Avoid nondeterministic (0, 0) output
-        ((2**31 - 1, zero_t), (0, 2**31 - 1)),  # Don't wrap to (0, 0) either
-        ((0, intmax_t), (0, 2**31 - 1)),  # Wrapping for the other argument
-    ]
-    for tc in test_cases:
-      tinput, toutput = tc[0], tc[1]
-      random_seed.set_random_seed(tinput[0])
-      g_seed, op_seed = data_random_seed.get_seed(tinput[1])
-      g_seed = self.evaluate(g_seed)
-      op_seed = self.evaluate(op_seed)
-      msg = 'test_case = {0}, got {1}, want {2}'.format(
-          tinput, (g_seed, op_seed), toutput)
-      self.assertEqual((g_seed, op_seed), toutput, msg=msg)
-      random_seed.set_random_seed(None)
+  @combinations.generate(
+      combinations.times(
+          test_base.default_test_combinations(),
+          _test_random_seed_combinations()
+      )
+  )
+  def testRandomSeed(self, input_fn, output_fn):
 
-    if not context.executing_eagerly():
-      random_seed.set_random_seed(1)
-      for i in range(10):
-        tinput = (1, None)
-        toutput = (1, i)
-        random_seed.set_random_seed(tinput[0])
-        g_seed, op_seed = data_random_seed.get_seed(tinput[1])
-        g_seed = self.evaluate(g_seed)
-        op_seed = self.evaluate(op_seed)
-        msg = 'test_case = {0}, got {1}, want {2}'.format(
-            1, (g_seed, op_seed), toutput)
-        self.assertEqual((g_seed, op_seed), toutput, msg=msg)
+    tinput, toutput = input_fn._obj(), output_fn._obj() # pylint: disable=protected-access
```

Comment:
Why is the private `_obj` API needed? The test combinations in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/data/experimental/kernel_tests/optimization/map_vectorization_test.py pass functions as parameters as well and do not need that API.

Reply:
This applies throughout this PR.

Reply:
I have removed the access to `_obj` wherever functions are being used within NamedObjects as calling NamedObject instances will internally call the function itself. However, in some cases, where `types` or `classes` are being layered with NamedObject's, the call to `_obj` is required to get hold of the actual data.

For instance, in `structure_test.py`:

```python
def testConvertLegacyStructure(self, output_types, output_shapes,
                                 output_classes, expected_structure):
    output_types = output_types._obj
    output_shapes = output_shapes._obj
    output_classes = output_classes._obj
```

Changes have been made accordingly.